packages/cerebral/src/tags/Tag.js

```diff
@@ -0,0 +1,100 @@
+/*
+  Creates tag for targetting things with a path in Cerebral
+*/
+export default class Tag {
+  constructor (tag, options = {}, strings, values) {
+    options.hasValue = options.hasValue === undefined ? true : options.hasValue
+
+    this.type = tag
+    this.options = options
+    this.strings = strings
+    this.values = values
+  }
+  /*
+    Returns all tags, also nested to identify nested state dependencies
+    in components
+  */
+  getTags (getters) {
+    if (!getters) {
+      throw new Error('You can not grab tags from a Tag without getters')
+    }
+
+    return [this].concat(this.getNestedTags(getters))
+  }
+  /*
+    Gets the path of the tag, where nested tags are evaluated
+  */
+  getPath (getters) {
+    if (!getters) {
+      throw new Error('You can not grab tags from a Tag without getters')
+    }
+
+    return this.populatePath(getters)
+  }
+  /*
+    Uses the path of the tag to look it up in related getter
+  */
+  getValue (getters) {
+    if (!getters) {
+      throw new Error('You can not grab tags from a Tag without getters')
+    }
+
+    if (this.options.hasValue && !getters[this.type]) {
+      throw new Error(`Tag of type ${this.type.toUpperCase()} can not be used in this context`)
+    }
+
+    if (this.options.hasValue) {
+      return typeof getters[this.type] === 'function' ? getters[this.type](this.getPath(getters)) : this.extractValueWithPath(getters[this.type], this.getPath(getters))
+    } else {
+      return this.getPath(getters)
+    }
+  }
+  /*
+    Extracts value from object using a path
+  */
+  extractValueWithPath (obj, path) {
+    return path.split('.').reduce((currentValue, key) => {
+      if (currentValue[key] === undefined) {
```

Comment:
The problem is when testing a value with code like this:

```js
when(input`foo`)
```

If foo is undefined, the code raises an issue. This could be fixed by having a special get for truthy/existence values, like "resolveArg.truthy()" and this code would not throw. Since we do not have this for now, throwing breaks code without any way to work around it.